Sigfox-sensit-v2.js

```diff
@@ -0,0 +1,378 @@
+  return BYTE3_FUNCTIONS[mode](byte3);
+}
+
+// Byte 3: Case for Temperature & Humidity, Magnet, Vibration or Button mode
+function byte3Case1(byte3) {
+  console.log('Byte 3: Case for Temperature & Humidity, Magnet, Vibration or Button mode');
+  const magnetState = byte3[1];
+  const temperatureLsb = byte3.slice(2, 8);
+  return { magnetState, temperatureLsb };
+}
+
+// Byte 3: Case for Light mode
+function byte3Case2(byte3) {
+  console.log('Byte 3: Case for Light mode');
+  const lightMask = byte3.slice(0, 2);
+  const lightValue = byte3.slice(2, 8);
+  return { lightMask, lightValue };
+}
+
+// Byte 3: Case for Door Opener mode
+function byte3Case3(byte3) {
+  console.log('Byte 3: Case for Door Opener mode');
+  const maxMagnet = byte3;
+  return { maxMagnet };
+}
+//---------------------------------------------------------------------
+
+// Byte 4 - variables extraction --------------------------------------
+
+// This byte depends on the mode
+function byte4VE(byte4, mode) {
+  console.log('Byte 4 VE ------------');
+  return BYTE4_FUNCTIONS[mode](byte4);
+}
+
+// Byte 4: Case for Temperature mode
+function byte4Case1(byte4) {
+  console.log('Byte 4: Case for Temperature mode');
+  const humidity = byte4;
+  return { humidity };
+}
+
+// Byte 4: Case for Magnet, Door opener, Light and Vibration mode
+function byte4Case2(byte4) {
+  console.log('Byte 4: Case for Magnet, Door opener, Light and Vibration mode');
+  const alertCounter = byte4; // Number of events that happened
+  return { alertCounter };
+}
+
+// Byte 4: Case for Button mode
+function byte4Case3(byte4) {
+  console.log('Byte 4: Case for Button mode');
+  const firmwareVMajor = byte4.slice(0, 4);
+  const firmwareVMinor = byte4.slice(4, 8);
+  return { firmwareVMajor, firmwareVMinor };
+}
+//---------------------------------------------------------------------
+
+// Variable decoding functions for each mode --------------------------
+
+// Only the temperature mode was implemented because missing information
+// about other modes.
+
+// Button mode - data decoding
+
+/*
+function mode_button (varValues)
+{
+    var firmwareVMajor = varValues[13];
+    var firmwareVMinor = varValues[14];
+} */
+
+// Temperature mode - data decoding
+function modeTemperature(varValues) {
+  console.log('Decoding temperature ----------------------');
+
+  // Expected input: [temperatureMsb, temperatureLsb, humidity]
+  let fTemperature;
+  let fHumidity; // Final variables
+
+  const temperatureMsb = varValues[0];
+  const temperatureLsb = varValues[1];
+  const humidity = varValues[2];
+
+  /*
+    console.log ("temperatureMsb: ", temperatureMsb);
+    console.log ("temperatureLsb: ", temperatureLsb);
+    console.log ("humidity: ", humidity);
+    */
+
+  fTemperature = `${temperatureMsb}${temperatureLsb}`;
+  console.log(fTemperature);
+  fTemperature = parseInt(fTemperature, 2);
+  console.log(fTemperature);
+  fTemperature = ((fTemperature - 200) / 8); // °C
+  console.log(fTemperature);
```

Comment:
you should print only the final temperature result, the previous calculus are useless